src/components/coin-icon/CoinIconFallback.js

```diff
@@ -22,65 +21,81 @@ const fallbackTextStyles = {
   textAlign: 'center',
 };
 
-const FallbackImage = styled(ImageWithCachedMetadata)(
-  ({
+const FallbackImage = React.memo(props => {
+  const {
     size,
-    theme: { colors },
     shadowColor: color,
     shadowOffset: { height: y, width: x },
     shadowOpacity: opacity,
     shadowRadius: radius,
     showImage,
-  }) => ({
-    height: size,
-    width: size,
-    ...position.coverAsObject,
-    ...shadow.buildAsObject(x, y, radius * 2, color, showImage ? opacity : 0),
-    backgroundColor: showImage ? colors.white : colors.transparent,
-    borderRadius: size / 2,
-    overflow: 'visible',
-  })
-);
+    as: ImageComponent = ImageWithCachedMetadata,
+  } = props;
+
+  const style = useMemo(
+    () => ({
+      height: size,
+      width: size,
+      ...position.coverAsObject,
+      ...shadow.buildAsObject(x, y, radius * 2, color, showImage ? opacity : 0),
+      backgroundColor: 'transparent',
+      borderRadius: size / 2,
+      overflow: 'visible',
+    }),
+    [size, showImage, x, y, radius, color, opacity]
+  );
 
-function WrappedFallbackImage({
-  color,
-  elevation = 6,
-  shadowOpacity,
-  showImage,
-  size,
-  eth,
-  ...props
-}) {
-  const { colors } = useTheme();
   return (
-    <Centered
+    <ImageComponent
       {...props}
-      {...position.coverAsObject}
-      {...borders.buildCircleAsObject(size)}
-      backgroundColor={colors.alpha(color || colors.dark, shadowOpacity || 0.3)}
-      elevation={showImage ? elevation : 0}
-      style={{ overflow: 'hidden' }}
-    >
-      <FallbackImage
-        as={eth ? Image : undefined}
-        source={EthIcon}
-        {...props}
-        overlayColor={color || colors.dark}
-        shadowOpacity={shadowOpacity}
-        showImage={showImage}
-        size={size}
-      />
-    </Centered>
+      cache={ImgixImage.cacheControl.immutable}
+      size={32}
+      style={style}
+    />
   );
-}
+});
+
+FallbackImage.displayName = 'FallbackImage';
+
+const WrappedFallbackImage = React.memo(
+  ({ color, elevation = 6, shadowOpacity, showImage, size, eth, ...props }) => {
+    const { colors } = useTheme();
```

Comment:
can we also pass "theme" down via props instead?

Reply:
I think so? How to find a balance between what to pass thru props and what to evaluate in components?